zerver/tests/test_message_edit.py

```diff
@@ -3793,6 +3793,47 @@ def test_move_message_to_stream_and_topic(self) -> None:
         )
         self.assert_json_success(result)
 
+    def test_move_many_messages_to_stream_and_topic(self) -> None:
+        (user_profile, old_stream, new_stream, msg_id, msg_id_later) = self.prepare_move_topics(
+            "iago", "first origin stream", "first destination stream", "first topic"
+        )
+
+        with queries_captured() as queries:
+            result = self.client_patch(
+                f"/json/messages/{msg_id}",
+                {
+                    "propagate_mode": "change_all",
+                    "send_notification_to_old_thread": "true",
+                    "stream_id": new_stream.id,
+                    "topic": "first topic",
+                },
+            )
+            self.assert_json_success(result)
+
+        # Adding more messages should not increase the number of
+        # queries
+        (user_profile, old_stream, new_stream, msg_id, msg_id_later) = self.prepare_move_topics(
+            "iago", "second origin stream", "second destination stream", "second topic"
+        )
+        for i in range(1, 201):
```

Comment:
Sending 200 messages as setup likely makes this test take a long time; can we get away with like 5?

Reply:
We need at least 101 to add an extra query, since this uses 100-message batches when calling `bulk_update`.  We can I guess make that a constant on the package so we can mock it out in testing?

Reply:
Actually, that doesn't make any sense, since we're removing the code that does that batching.  This number needs to be large enough that, _if we were doing batches_ rather than one statement, it would be larger than the batch size.

Reply:
We should be able to override the batch size for the purposes of a test.

Reply:
That said, this also seems like a detail we could file a follow-up issue for fixing in order to integrate the rest.